### ISSUE TITLE
Question: why do files on my NAS have creation date 9 hours earlier?

### ISSUE THREAD

**Issue by mvanbreeden** [ccrisan/motioneye#86]
My NAS has proper time settings, the Raspberry does so also. Yet movies created have a creation date 9 hours earlier (I'm in Amsterdam timezone). I do not seem to find why...

**Comment by ccrisan** [comment]
How about the timestamp that's overlaid on the video preview? Is it wrong as well? What have you set for Image File Name and Movie File Name.

**Comment by mvanbreeden** [comment]
The overlaid timestamp is correct, and so are file creation timings for thumbs. I did not change defaults for filenames, which show correct time as well. It's just file creation data for avi's on NAS that's wrong

**Comment by ccrisan** [comment]
Hmm well no idea about that. I never cared for the ctime/mtime attributes of files as they tend to be misleading. Are you running Raspbian? Have you rebooted since you changed your timezone?

**Comment by mvanbreeden** [comment]
Yes I have rebooted, and running Raspbian. It just struck me that's all,
when looking in Finder. I'll check in the source myself, will keep you
posted. No biggie.

Being new to github, I wasn't able to label the issue as question... Sorry
for that
Op do 31 dec. 2015 om 11:57 schreef Calin Crisan <notifications@github.com>

**Comment by ccrisan** [comment]
Yes, please do :)

**Comment by mvanbreeden** [comment]
I found out what the problem is,  will think about solution.
The problem is that the subprocess.open call for ffmpeg (make_movie) does not contain environment set. It assumes that environment variables will be gotten from parent process, but somehow those (or at least TZ) is not there.
The call to ffmpeg in make_movie_preview (check_out instead of Popen) does give environment info.

So probably they are called from different parent processes. Do you have any clue already? If not, I'll dive a little deeper ;-)

**Comment by mvanbreeden** [comment]
I was ging about this wrong earlier. The problem with timezone is with starting motion with subprocess.popen, which apparantly does not give motion the right environment. Therefore motion will save the files without timezone info.

The best way to solve would be to exec as shell, what do you think @ccrisan?

**Comment by ccrisan** [comment]
It's a valid question that you're rising here. I'll have to look into it and see how exactly the timezone is passed onto subprocesses. As far as I know, glibc will look for the /etc/zoneinfo file to deduce that information; the environment shouldn't have anything to do with it. But then again, I'm not sure about it and I'll have to run some tests.

**Comment by JamBalaya56562** [comment]
The key detail here is that the overlaid timestamp and the file names are correct — that means `motion` itself has the right local time. What's off is only the file's `ctime`/`mtime` on the NAS.

Those attributes don't store a timezone: they're a plain Unix epoch (effectively UTC). Whatever displays them (Finder, the NAS UI, `ls`) converts that epoch to a timezone at display time. So a fixed 9-hour offset almost always means the *viewer* or the *NAS* is interpreting the timestamp in a different timezone than the Raspberry Pi that wrote it — not that motionEye wrote a wrong value. The old subprocess-environment theory earlier in this thread doesn't really fit, since a bad `TZ` for `motion` would also have corrupted the overlay/filenames, which were fine.

Worth checking:
- the timezone configured on the **NAS** itself, and the client you're browsing from (Finder shows dates in the client's tz);
- for SMB/NFS shares, how the mount maps ownership/times;
- if you're on a **current Docker install**, make sure the container has the timezone set — pass `-e TZ=Europe/Amsterdam` and/or mount `-v /etc/localtime:/etc/localtime:ro`, otherwise the container runs in UTC and files land with UTC times.

This one is from 2015 and predates the Python 3 rewrite, so the internals have changed a lot. **This one looks safe to close** — if it's still reproducible on a current motionEye release, a fresh report with the version and your NAS/mount setup would be the way to look again.

**Comment by MichaIng** [comment]
Thanks for detailing the most likely explanation. I'll close this old question.
